src/ApiFilter.php

```diff
@@ -2,6 +2,111 @@
+
+        $this->registerApplicator(new ApplicatorSql(), Priority::MEDIUM);
+
+        $this->registerEscape(new EscapeInt(), Priority::MEDIUM);
+        $this->registerEscape(new EscapeString(), Priority::LOW);
+    }
+
+    /**
+     * Parse query parameters into FiltersInterface, which can be applied later on some `filterable`
+     *
+     * @example
+     * With Symfony
+     * $filters = $apiFilter->parseFilters($request->query->all())
+     */
+    public function parseFilters(array $queryParameters): FiltersInterface
+    {
+        return $this->parser->parse($queryParameters);
+    }
+
+    /**
+     * Apply one Filter to given filterable and returns the result of the same type as given filterable
+     * or whatever the Applicator returns
+     *
+     * Filterable might be anything, but there must be an Applicator for that filterable
+     * First Applicator (from the highest priority), which can be applied is applied and no others
+     * @see ApplicatorInterface
+     *
+     * You can register you own Applicators
+     * @see ApiFilter::registerApplicator()
+     *
+     * @param mixed $filterable
+     * @return mixed
+     */
+    public function applyFilter(FilterInterface $filter, $filterable)
+    {
+        return $this->applicator->apply($filter, new Filterable($filterable))->getValue();
+    }
+
+    /**
+     * Apply all Filters to given filterable and returns the result of the same type as given filterable
+     * or whatever the Applicator returns
+     *
+     * Filterable might be anything, but there must be an Applicator for that filterable
+     * First Applicator (from the highest priority), which can be applied is applied and no others
+     * @see ApplicatorInterface
+     *
+     * You can register you own Applicators
+     * @see ApiFilter::registerApplicator()
+     *
+     * @param mixed $filterable
+     * @return mixed
+     */
+    public function applyFilters(FiltersInterface $filters, $filterable)
+    {
+        return $this->applicator->applyAll($filters, new Filterable($filterable))->getValue();
+    }
+
+    /**
+     * Add another applicator which will be try to use on apply filter(s)
+     * First Applicator (from the highest priority), which can be applied is applied and no others
+     *
+     * Priority can be any integer value (or use predefined Priority)
+     * @see Priority
+     */
+    public function registerApplicator(ApplicatorInterface $applicator, int $priority): void
+    {
+        $this->applicator->registerApplicator($applicator, $priority);
+    }
+
+    /**
+     * Add another escape which will be try to use on values before while applying the filter
+     * First Escape (from the highest priority), which can be applied is applied and no others
+     *
+     * Priority can be any integer value (or use predefined Priority)
+     * @see Priority
+     */
+    public function registerEscape(EscapeInterface $escape, int $priority): void
```

Comment:
I would rename this everywhere to "Escaper". `registerEscaper()`, `addEscaper`, `$escaperService`, `EscaperInterface` etc. to make it noun - similarly as "Applicator".

Reply:
Yes it makes sense, but as I said before, I would really like to get rid of escaping like this at all, so I dont want to rename all of these now and remove them in the next PR. So I might add a note to `todo` list - `either remove escapers or rename them to escapers`, is it ok?